sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -150,6 +150,57 @@ object IntervalUtils {
     }
   }
 
+  private val daySecondStringPattern = ("(?i)^(INTERVAL\\s+)([+|-])?(')" +
+    "([+|-])?(\\d+) (\\d{1,2}):(\\d{1,2}):(\\d{1,2})(\\.\\d{1,9})?(')(\\s+DAY\\s+TO\\s+SECOND)$").r
+  private val daySecondPattern = "^([+|-])?(\\d+) (\\d{1,2}):(\\d{1,2}):(\\d{1,2})(\\.\\d{1,9})?$".r
```

Comment:
Let's extract the common pattern:
```suggestion
  private val unquotedDaySecondPattern =
    "([+|-])?(\\d+) (\\d{1,2}):(\\d{1,2}):(\\d{1,2})(\\.\\d{1,9})?"
  private val quotedDaySecondPattern = (s"^$unquotedDaySecondPattern$$").r
  private val daySecondLiteralPattern =
    (s"(?i)^(INTERVAL\\s+)([+|-])?(')$unquotedDaySecondPattern(')(\\s+DAY\\s+TO\\s+SECOND)$$").r
```